network/swarm.py

```diff
@@ -19,28 +19,24 @@ def __init__(self, my_peer_id, peerstore, upgrader):
         self.transport = None
 
     def get_peer_id(self):
-        return self.id
+        return self.peer_id
 
-    def set_stream_handler(self, protocol_id, stream_handler):
-        """
-        :param protocol_id: protocol id used on stream
-        :param stream_handler: a stream handler instance
-        :return: true if successful
+    async def dial_peer(self, peer_id):
         """
-        self.stream_handlers[protocol_id] = stream_handler
+        dial_peer try to create a connection to peer_id
 
-    async def new_stream(self, peer_id, protocol_id):
-        """
-        :param peer_id: peer_id of destination
-        :param protocol_id: protocol id
-        :return: net stream instance
+        :param peer_id: peer if we want to dial
+        :raises SwarmException: raised when no address if found for peer_id
+        :return: muxed connection
         """
+
         # Get peer info from peer store
         addrs = self.peerstore.addrs(peer_id)
 
         if not addrs:
             raise SwarmException("No known addresses to peer")
 
+        # TODO: define logic to choose which address to use, or try them all ?
```

Comment:
I believe we should try them all and use whichever address responds the fastest

Reply:
This is what the go-libp2p repo is doing at the moment, and there are plans to upgrade it to a smarter dial. I forgot if it's called a traffic shaper or a dial manager. There is a bit of design freedom here. We should participate in the discussion with the rest of the community.

Reply:
@zixuanzh do you have a link to the issue where they discuss this dial manager ?

Reply:
this is the thread https://github.com/libp2p/go-libp2p-swarm/pull/88, dial manager might be a make-up name.